rfcs/theming-color-scheme-strategy.md

```diff
@@ -0,0 +1,168 @@
+
+#### Utility variables
+
+Utility variables are variables that helps crafting experiences, while being utilitarian. These are not values that have been discussed with designers, a circle being a circle there is no art direction on these. Creating a circle with a border-radius is a common CSS trick that is used all over Orbit and the web in general. Instead of repeating the same line of CSS all the time, there is an utility for that. Many more utilities are available through Orbit, check Orbit's documentation for more information.
+
+```css
+.o-ui {
+  --o-ui-circular-border-radius: 100%;
+}
+
+// ...
+
+.circle {
+    border-radius: var(--o-ui-circular-border-radius);
+}
+```
+
+### Implementation in Orbit
+
+The *Theme Provider* componen is responsible of adding the 3 needed classes: `-o-ui o-ui-{theme} -o-ui-{color-scheme}) in order to benefit from the complete Orbit UI experience.
+
+Orbit's utility variables are declared at the root CSS file `react-components/src/index.css` as well as generic CSS declarations like the app default font-color.
+
+Global variables, alias variables as well as alias CSS classes are declared at the foundation level, grouped by what they affect:  background, color, spacing, etc. Each of these files contains global declarations as well as theme and color scheme declarations. Here's is an example of the structure of a CSS file.
+
+```css
+// background.css
+.o-ui {
+  --o-ui-global-background-...: value;
+}
+
+.o-ui-light-theme {
+  --o-ui-alias...: value;
+}
+
+.o-ui-dark-theme {
+  --o-ui-alias...: value;
+}
+
+.o-ui-apricot {
+  --o-ui-alias...: value;
+}
+
+.o-ui-alias-... {
+  background-color: var(--o-ui-alias...);
+}
+
+...
+```
+
+Concretely here are the files that holds all of Orbit CSS
+
+- colors : all of Orbit's colors
+- background : aliases for our backgrounds
+- text: aliases for our text colors
+- lines: aliases for our borders
+- icons: aliases for our icon fills
+- shadows: shadow definitions as well as their aliases
+- spacing / type scale
+
+### Implementation in apps
+
+In order to be able to benefits from themes, an app need some setup. Apart from having Orbit installed, see Orbit documentation, 3 classes are needed at the root element of your app, Orbit ThemeProvider takes care of this. Here is the classes provided along with what they do
+
+- `o-ui` gives you access to design tokens via CSS Custom Properties
+- `o-ui-dark` `o-ui-light` gives you access to semantic tokens via CSS Custom Properties
+- `o-ui-apricot` `o-ui-desktop` gives you access to brand related semantic tokens, e.g. `--o-ui-primary-50`
+
+Once setup you can refer to these variables in your CSS, as well as use their utility classes counterpart.
+
+Using a semantic token in your CSS
```

Comment:
We should probably have some kind of ThemeProvider example? Basically, for an app, the ThemeProvider is the root of all this.